.github/workflows/release-please.yml

```diff
@@ -0,0 +1,39 @@
+name: release-please
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v2
+        id: release
+        with:
+          package-name: node-gyp
+          release-type: node
+          changelog-types: >
+            [{"type":"feat","section":"Features","hidden":false},
+            {"type":"fix","section":"Bug Fixes","hidden":false},
+            {"type":"bin","section":"Core","hidden":false},
+            {"type":"gyp","section":"Core","hidden":false},
+            {"type":"lib","section":"Core","hidden":false},
+            {"type":"src","section":"Core","hidden":false},
+            {"type":"test","section":"Tests","hidden":false},
+            {"type":"build","section":"Core","hidden":false},
+            {"type":"configure","section":"Core","hidden":false},
+            {"type":"install","section":"Core","hidden":false},
+            {"type":"deps","section":"Core","hidden":false},
+            {"type":"python","section":"Core","hidden":false},
+            {"type":"win","section":"Core","hidden":false},
+            {"type":"zos","section":"Core","hidden":false},
+            {"type":"doc","section":"Doc","hidden":false},
+            {"type":"docs","section":"Doc","hidden":false},
+            {"type":"readme","section":"Doc","hidden":false},
+            {"type":"chore","section":"Miscellaneous","hidden":false},
+            {"type":"refactor","section":"Miscellaneous","hidden":false},
+            {"type":"ci","section":"Miscellaneous","hidden":false},
+            {"type":"travis","section":"Miscellaneous","hidden":false},
```

Comment:
Travis looks redundant here

Reply:
Thanks for the feedback.

Travis makes sense to remove, especially since we are only using GitHub Actions now. It's not hurting anything, I think, but it is odd to see it there.

---

This list is the full set of prefixes that will be included in (read: won't be omitted from) the Changelog.

I expect people _won't_ use _most_ of these, but commit prefixes that `release-please` doesn't recognize are left out of the Changelog entirely. I added almost every prefix I saw in the commit history, going all the way back to the first commit. My thinking was "the more the better," for comprehensive coverage. I didn't want anything that is merged to be left out of the Changelog.

On the other hand, leaving certain prefixes out of the list would be a way to intentionally exclude them from the final Changelog. If that would be preferred, (to "reduce clutter" in the Changelog, I suppose?), then I can remove some more of these. (For example: the default for `chore:` is exclusion from the Changelog, but now with `chore:` on the list, those commits _will_ be included. I thought including them was better.)

Reply:
Ones I missed or decided not to include:
- `tools:` (used for some small `gyp` tweaks back in 2018/2019: https://github.com/nodejs/node-gyp/commit/ea9aff44f20e9272e173ebefa861e6920cd9f73a, https://github.com/nodejs/node-gyp/commit/e8ea74e0fa410bbde1ca8519f5e3da68d4c87052, https://github.com/nodejs/node-gyp/commit/6f5a408934d4f89b2ee1923a11ff44a93d0d44a1. Two of these patches were ported from `nodejs/node`'s `tools/gyp`. the original prefix from `nodejs/node` was preserved.)
- `History:` (apparently CHANGELOG.md used to be History.md. https://github.com/nodejs/node-gyp/commit/89fe78199b017807122ea1a758bda503d4ba634c)
- `addon:` (another small `gyp` tweak: https://github.com/nodejs/node-gyp/commit/be4ea598ddf23034fcf7c7b163b5829ad771c7b2)
- `rebuild:` (adjustments to the `node-gyp rebuild` subcommand: https://github.com/nodejs/node-gyp/commit/2f675ab223e8ee191d6e191276e4275bc6f1e80c, https://github.com/nodejs/node-gyp/commit/0eacc90adcf7e7154c16bc5a4b7e4b435ea0caf0)
- `package:` (Various updates to `package.json`, e.g.: https://github.com/nodejs/node-gyp/commit/d535620c86f3ab874202821a1d55c25151fc8208. This doesn't seem like a good prefix to use, in my opinion.)
- `nopt:` (Widening the version range for the `nopt` package in `package.json`... https://github.com/nodejs/node-gyp/commit/a93752fa73d4968b4fb5f9dc08230578f0c63a2d.)
- `node-gyp:` (Updates to the `lib/node-gyp.js` file: https://github.com/nodejs/node-gyp/commit/e82e4387bed35cc59d61a6433dc23d9caf0bdc01, https://github.com/nodejs/node-gyp/commit/b4302a3db27831fc80ece98d61a4e5b2bb7f1757, https://github.com/nodejs/node-gyp/commit/c4bb689abb1f9410e33fa060608edddcb64a4750)
- `xcode_emulation:` (A patch cherry-picked from upstream `gyp` that allows running on macOS with "command line tools" rather than full Xcode. The original prefix from upstream was preserved: https://github.com/nodejs/node-gyp/commit/5785bae9b00c8324d8df9cd87aa4ac2531f1da72)
- `node_modules:` (pruning unnecessary dirs from the (at that time checked-in) `node_modules` dir: https://github.com/nodejs/node-gyp/commit/484b2866d415a47258de82479472a91337c07040)
- `addon.gypi:` (adjustments to the `addon.gypi` file)
- `remove` (the `remove` subcommand of `node-gyp` (`lib/remove.js`): 800179a, 119e030)
- `legacy:` (an old `legacy` dir, now removed.)
- `jshintrc:` (An old `.jshintrc` config file, now removed: b3a147e)
- `windows:` (fully-spelled-out `win:`: 23bd601)
- `list:` (The `list` subcommand of `node-gyp` (`lib/list.js`): ae4d8d1e7940f8059708c8d6327ffbaa1cd163a0)
- `core:` (Various updates to the files under `lib/`: a185536, ae39e1a, 62e5851, 1354d27)
- `'clean':` (adjustment to the `clean` subcommand of `node-gyp`... quoted for some reason: 06aac83)

---

Of these, I think I want to add:
- The rest of the subcommands (`clean:`, `list:`, `rebuild:`, `remove:`).
  - And to complete the `lib/` folder set, maybe `node-gyp:` for `lib/node-gyp.js`, but I think we should really use `lib:` for that instead.
- Probably add `windows:`, which in turn makes me want to add `mac:` and `macOS:`, `linux:` and maybe `lin:`.

Reply:
Updated.

Although I think maybe encouraging people to just use `chore:` `fix:` and `feat:` might be simpler and for the best. In which case dropping all the custom configured prefixes might be the most appropriate move. (And consider adding a bot that enforces Conventional Commits commit messages? Which might annoy/drive away new contributors... maintainer discipline in what messages are used when landing commits would also take care of the problem.)